internal/task/task/common/client_status.go

```diff
@@ -81,11 +93,76 @@ func (s *step2FormatClientStatus) Execute(ctx *context.Context) error {
 		Kind:        client.Kind,
 		ContainerId: s.containerId,
 		Status:      status,
+		Address:     address,
 		AuxInfo:     client.AuxInfo,
 	})
 	return nil
 }
 
+func (s *step2GetAddress) Execute(ctx *context.Context) error {
+	cmd := ctx.Module().DockerCli().TopContainer(s.containerId)
+	out, err := cmd.Execute(s.execOptions)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(out, "\n")
+	var pid string
+	if len(lines) > 1 {
+		reg := regexp.MustCompile(`\s+`)
+		res := reg.Split(lines[1], -1)
+		if len(res) > 1 {
+			pid = res[1]
+		}
+	}
+
+	if len(pid) == 0 {
+		return nil
+	}
+
+	// execute "ss" command in container
+	cli := ctx.Module().Shell().SocketStatistics("")
+	cli.AddOption("--no-header")
+	cli.AddOption("--processes")
+	cli.AddOption("--listening")
+	command, err := cli.String()
+	if err != nil {
+		return err
+	}
+
+	cmd = ctx.Module().DockerCli().ContainerExec(s.containerId, command)
+	out, err = cmd.Execute(s.execOptions)
+	if err != nil {
+		return err
+	}
+
+	// handle output
+	lines = strings.Split(out, "\n")
+	for _, line := range lines {
+		address := s.extractAddress(line, pid)
+		if len(address) > 0 {
+			*s.address = address
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// e.g: tcp LISTEN 0 128 10.246.159.123:2379 *:* users:(("etcd",pid=7,fd=5))
+// e.g: tcp LISTEN 0 128 *:2379 *:* users:(("etcd",pid=7,fd=5))
+func (s *step2GetAddress) extractAddress(line, pid string) string {
+
+	regex, err := regexp.Compile(`^.* ((\d+\.\d+\.\d+\.\d+)|\*:\d+).*pid=` + pid + ".*$")
+	if err == nil {
+		mu := regex.FindStringSubmatch(line)
+		if len(mu) > 1 {
+			return mu[1]
+		}
+	}
+	return ""
+}
```

Comment:
newer version ss command will answer `RTNETLINK answers: Invalid argument`. See the pr https://github.com/opencurve/curveadm/commit/5adc604ae3cedd156171da4006670b05fbb04ecd